### PR TITLE
[lerc] Update to 3.0, move config to unofficial

### PR DESCRIPTION
**Pull request by dg0yt**
- #### What does your PR fix?
  Updates LERC to v3.0.
  Modernizes the portfile.
  Install `NOTICE` as required by license.
  Adds the include path to exported target.
  **Breaking change:** Upstream changed the library target name from `LercLib` to `Lerc`. I didn't add polyfill, but took this opportunity to move the exported target (added by vcpkg) to the `unofficial::Lerc::` namespace and to rename the package to `unofficial-lerc`. I guess it wasn't a guideline at the time the patch was added, but it is now.
  (Adding polyfill would also require adding an explicit `usage` file.)
  CC @n4z4m3

  This version is needed for de-vendored LERC support in GDAL 3.5.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  unchanged, no

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  yes.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/lerc/portfile.cmake [hunk 1]
```diff
@@ -1,24 +1,20 @@
...
     OUT_SOURCE_PATH SOURCE_PATH
     REPO Esri/lerc
-    REF  v2.2
-    SHA512 5ddf1e8f0c123d3c1329e980021e25e6ff9b79c96588115e5b48ba7637f0b2bf3ebb2ab6ebf94cfbde45ea1521f14405f669e23f0b74d9ae8f9b2cf80a908215
+    REF  v3.0
+    SHA512 8e04d890c4d46528641b354ec3f47f2b0563e1740927ac894925a30f7de2b235cb3517ce7e477886bd3eb50ebd6c6e78f22d73d5500877e552d9e684e626293b
     HEAD_REF master
     PATCHES
-        "install_lib_to_archive_path.patch"
-        "enable_static_build.patch"
```
Comment (JonLiu1993) [inline ports/lerc/portfile.cmake:9]: These patches apply to the latest version

**Review by JonLiu1993 — APPROVED** [review]

File: ports/lerc/portfile.cmake (hunk 1 above)
Comment (JonLiu1993) [reply ports/lerc/portfile.cmake:9]: > > These patches apply to the latest version
> 
> These patches "don't apply to the latest version" but are included upstream.

@dg0yt ,Sorry, I made a typo, I meant that the patches have been applied to the latest version

**Comment by JonLiu1993** [comment]
I tested the usage successfully
````
1> CMake generation started for configuration: 'x64-Debug'.
1> Command line: "C:\WINDOWS\system32\cmd.exe" /c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2019\ENTERPRISE\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Visual Studio 16 2019" -A x64  -DCMAKE_CONFIGURATION_TYPES:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="C:\Users\source\repos\CMakeProject42\out\install\x64-Debug" -DCMAKE_TOOLCHAIN_FILE=F:/Feature-test/lerc/vcpkg/scripts/buildsystems/vcpkg.cmake "C:\Users\source\repos\CMakeProject42" 2>&1"
1> Working directory: C:\Users\source\repos\CMakeProject42\out\build\x64-Debug
1> [CMake] -- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19042.
1> [CMake] -- The C compiler identification is MSVC 19.29.30137.0
1> [CMake] -- The CXX compiler identification is MSVC 19.29.30137.0
1> [CMake] -- Detecting C compiler ABI info
1> [CMake] -- Detecting C compiler ABI info - done
1> [CMake] -- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
1> [CMake] -- Detecting C compile features
1> [CMake] -- Detecting C compile features - done
1> [CMake] -- Detecting CXX compiler ABI info
1> [CMake] -- Detecting CXX compiler ABI info - done
1> [CMake] -- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
1> [CMake] -- Detecting CXX compile features
1> [CMake] -- Detecting CXX compile features - done
1> [CMake] -- Configuring done
1> [CMake] -- Generating done
1> [CMake] -- Build files have been written to: C:/Users/source/repos/CMakeProject42/out/build/x64-Debug
1> Extracted CMake variables.
1> Extracted source files and headers.
1> Extracted code model.
1> Extracted toolchain configurations.
1> Extracted includes paths.
1> CMake generation finished.

````

**Comment by BillyONeal** [comment]
>I guess it wasn't a guideline at the time the patch was added, but it is now.

Yeah, it's one not all the maintainers properly understood because we didn't write it down :(. I've been trying to make sure such tribal knowledge is in the maintainer guide, sorry!

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**